rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -913,6 +913,26 @@ object Reduce {
         }
     }
 
+    private[this] val toUtf8Bytes: Method = new Method() {
+
+      override def apply(
+          p: Par,
+          args: Seq[Par]
+      )(implicit env: Env[Par], costAccountingAlg: CostAccountingAlg[M]): M[Par] =
+        if (args.nonEmpty) {
+          s.raiseError(ReduceError("Error: toUtf8Bytes does not take arguments"))
+        } else {
+          p.singleExpr() match {
+            case Some(Expr(GString(encoded))) =>
+              for {
+                _ <- costAccountingAlg.charge(hexToByteCost(encoded))
+              } yield Expr(GByteArray(ByteString.copyFrom(encoded.getBytes("UTF-8"))))
+            case _ =>
+              s.raiseError(ReduceError("Error: toUtf8Bytes can be called only on single strings."))
```

Comment:
For these kind of message use `MethodNotDefined`.

Reply:
I am NOT a scala dev, so, I appologized for needing to ask basic questions, refactoring to use `MethodNotDefined`, can I do the following:
```
          p.singleExpr() match {
            case Some(Expr(GString(encoded))) =>
              for {
                _ <- costAccountingAlg.charge(hexToByteCost(encoded))
              } yield Expr(GByteArray(ByteString.copyFrom(encoded.getBytes("UTF-8"))))
            case Some(Expr(other)) =>
              s.raiseError(MethodNotDefined("toUtf8Bytes", other.typ))
          }
```